pkg/kubectl/ingress.go

```diff
@@ -0,0 +1,176 @@
+		delegate.Name = name
+	} else {
+		return nil, fmt.Errorf("expected string, saw %v for 'name'", params["name"])
+	}
+
+	if host, isArray := params["host"].([]string); isArray {
+		delegate.Host = host
+	} else {
+		return nil, fmt.Errorf("expected []string, found %v for 'host'", params["host"])
+	}
+
+	if params["tls-acme"] != nil {
+		if tlsAcme, isBool := params["tls-acme"].(bool); isBool {
+			delegate.TLSAcme = tlsAcme
+		} else {
+			return nil, fmt.Errorf("expected bool, saw %v for 'tls-acme'", params["tls-acme"])
+		}
+	}
+
+	if params["service-name"] != nil {
+		if serviceName, isString := params["service-name"].(string); isString {
+			delegate.ServiceName = serviceName
+		} else {
+			return nil, fmt.Errorf("expected string, saw %v for 'service-name'", params["service-name"])
+		}
+	}
+
+	if params["service-port"] != nil {
+		if servicePortInt, isInt := params["service-port"].(int); isInt {
+			delegate.ServicePort = intstr.FromInt(servicePortInt)
+		} else if servicePortString, isString := params["service-port"].(string); isString {
+			delegate.ServicePort = intstr.FromString(servicePortString)
+		} else {
+			return nil, fmt.Errorf("expected string or int, saw %v for 'service-port'", params["service-port"])
+		}
+	}
+
+	return delegate.StructuredGenerate()
+}
+
+// StructuredGenerate outputs a ResourceQuota object using the configured fields
+func (g *IngressV1Beta1) StructuredGenerate() (runtime.Object, error) {
+	if err := g.validate(); err != nil {
+		return nil, err
+	}
+
+	// setup default label and selector
+	labels := map[string]string{}
+	labels["app"] = g.Name
+
+	ingress := extensions.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   g.Name,
+			Labels: labels,
+		},
+	}
+
+	// Default backend service name to same name as ingress
+	backendServiceName := g.ServiceName
+	if backendServiceName == "" {
+		backendServiceName = g.Name
+	}
+
+	// Default backend service port to 80
+	backendServicePort := g.ServicePort
+	if backendServicePort.IntVal == 0 && backendServicePort.StrVal == "" {
+		backendServicePort.IntVal = 80
+	}
+
+	for _, host := range g.Host {
+		rule := extensions.IngressRule{
+			Host: host,
+		}
+		rule.HTTP = &extensions.HTTPIngressRuleValue{
+			Paths: []extensions.HTTPIngressPath{
+				{
+					Path: "/",
+					Backend: extensions.IngressBackend{
+						ServiceName: backendServiceName,
+						ServicePort: backendServicePort,
+					},
+				},
+			},
+		}
+		ingress.Spec.Rules = append(ingress.Spec.Rules, rule)
+	}
+
+	if g.TLSAcme {
+		tls := extensions.IngressTLS{
+			Hosts:      g.Host,
+			SecretName: "tls-" + g.Name,
+		}
+		ingress.Spec.TLS = append(ingress.Spec.TLS, tls)
+
+		if ingress.Annotations == nil {
+			ingress.Annotations = make(map[string]string)
+		}
+		ingress.Annotations[ingressAnnotationNameTLSAcme] = "true"
+	}
+
+	return &ingress, nil
+}
+
+// validate validates required fields are set to support structured generation
+func (g *IngressV1Beta1) validate() error {
```

Comment:
If `service-port` is specified, should we validate if the `service-port` has a valid name or be in a valid range?